clients/src/test/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumerTest.java

```diff
@@ -1835,7 +1849,31 @@ public void testPollThrowsInterruptExceptionIfInterrupted() {
         }
         assertDoesNotThrow(() -> consumer.poll(Duration.ZERO));
     }
-    
+
+    @Test
+    void testReaperInvokedInClose() {
+        consumer = newConsumer();
+        consumer.close();
+        verify(backgroundEventReaper).reap(backgroundEventQueue);
+    }
+
+    @Test
+    void testReaperInvokedInUnsubscribe() {
+        consumer = newConsumer();
+        completeUnsubscribeApplicationEventSuccessfully();
+        consumer.unsubscribe();
+        verify(backgroundEventReaper).reap(any(Long.class));
```

Comment:
You control the time here. Why do you not verify that `reap()` is called with the correct time?

Reply:
Good call. Done!